src/Components/Square.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import {SquareProps} from '../Types/TypeProps'
+
+/**
+ * Render board cell buttons
+ * @param {*} props (Action properties from parent(<Game />) component)
+ */
+const Square: React.FC<SquareProps> = (props: SquareProps) => {
+
+  /**
+   * Handle action for each cell when clicked
+   * (This method was basically created to circumvent issue with passing down parameters)
+   * @param event 
+   */
+  const extendHandleClick = (event:any) => {
```

Comment:
I would more strictly type this, you'll have to update your other types though:

```suggestion
  const extendHandleClick = (event: React.MouseEventHandler<HTMLButtonElement>) => {
```